docs/wallet_browser_extension/provider/api_documentation.md

```diff
@@ -0,0 +1,430 @@
+---
+tags:
+  - wallet
+  - browser extension
+  - provider API
+---
+
+# API Documentation
+
+## Properties
+
+### isXPLL
+
+This property is true if the user has extension installed.
+
+```js
+window.xpll.isXPLL; // => true
+```
+
+## Request
+
+The request method is intended as a transport- and protocol-agnostic wrapper function for [Remote Procedure Calls (RPCs)][1]
+
+If resolved, the Promise will resolve with a result per the RPC method’s specification.
+If the returned Promise rejects,
+it will reject with a `XPLLProviderRpcError` as specified in the [Errors][2] section below.
+
+```ts
+interface RequestArguments {
+  readonly method: string;
+  readonly params?: readonly unknown[] | object;
+}
+```
+
+```ts
+request(data: RequestArguments): Promise<Result>;
+// or reject with `XPLLProviderRpcError`
+```
+
+### Parameters
+
+- `method`: Method indicates which RPC method to call.
+- `params`: Parameters is an optional array or object of arguments to pass with the RPC.
+
+### Helper Functions
+
+To improve the developer experience,
+the provider also implemented a list of helper functions,
+which helps the user simplify the RPC method call.
+
+## Request Permission
+
+Request [permission][4] to access the user's accounts.
+
+!!! warning
+    The request causes a popup to appear.
+    You should only request permissions in response to a direct user action, such as a button click.
+
+!!! failure
+    If user reject the request, the `User Rejected Request 4001` error will be thrown.
+
+```ts
+interface Permission {
+  id: string;
+  invoker: string;
+  parent_capability: string;
+  caveats: {
+    type: string;
+    value?: unknown;
+  }[];
+  created_at: number;
+}
+interface RequestPermissionRequest {
+  method: "request_permissions";
+}
+type RequestedPermissionResult = Permission;
+```
+
+=== "RPC"
+    ```js
+    window.xpll.request({
+      method: "request_permissions";
+    });
+    ```
+
+=== "Helper Function"
+    ```js
+    window.xpll.requestPermissions();
+    ```
+
+## Get Permissions
```

Comment:
The difference between "Request Permission" and "Get Permission" is unclaer

Reply:
solved by 59a14af